src/invoice.rs

```diff
@@ -208,6 +244,33 @@ impl FromStr for RgbInvoice {
                 }
             };
 
+        let mut query: IndexMap<String, String> = IndexMap::new();
+        if let Some(q) = uri.query() {
+            if let Ok(query_str) = q.decode().into_string() {
+                let params = query_str.split('&');
+                for p in params {
+                    let mut kv = p.split('=');
+                    if kv.clone().count() != 2 {
+                        return Err(InvoiceParseError::InvalidQueryParam);
+                    };
+                    let (k, v) = (kv.next().unwrap(), kv.next().unwrap());
+                    query.insert(k.to_string(), v.to_string());
+                }
+            } else {
+                return Err(InvoiceParseError::Invalid);
+            }
+        }
+
+        let mut expiry = None;
+        if let Some(exp) = query.iter().find(|(k, _)| *k == "expiry") {
```

Comment:
`"expire"` string is better to put into a constant since it is used in several places and if we will even change the name, we may forget about other occurrences.

Reply:
yep, forgot to do that. updated